### ISSUE TITLE
VoiceOver Identifies Date Segment as stepper, but increment/decrement Controls does not work

### ISSUE THREAD

**Issue by jafar-saadique-klara** [adobe/react-spectrum#8591]
### Provide a general summary of the issue here

VoiceOver incorrectly identifies date segment as stepper, but `Control + Option + Up/Down Arrow` does not change the value

### 🤔 Expected Behavior?

When using VoiceOver on macOS, if a date segment (such as day, month, or year) is identified as a stepper, pressing `Control + Option + Up Arrow` should increase the value, and `Control + Option + Down Arrow` should decrease the value accordingly

### 😯 Current Behavior

VoiceOver identifies the date segment as a stepper and announces that the value can be changed using `Control + Option + Up/Down Arrow`. However, pressing these key combinations does not actually change the value. The interaction appears unresponsive, and the date segment remains unchanged

### 💁 Possible Solution

Explicitly implementing accessibility increment/decrement behaviour in the component so that VoiceOver can trigger value changes as expected when using `Control + Option + Up/Down Arrow`

### 🔦 Context

For the native date input, Voiceover correctly identifies it as stepper and `Control + Option + Up/Down Arrow` works
https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Elements/input/date

### 🖥️ Steps to Reproduce

1. Open the following component in Safari on macOS: https://react-spectrum.adobe.com/react-aria/DateField.html
2. Enable VoiceOver (Command + F5)
3. Navigate to the date segment
4. VoiceOver announces the segment as a stepper and says you can change the value using `Control + Option + Up/Down Arrow`
5. Attempt to use `Control + Option + Up/Down Arrow`
6. Observe that the value does not change

### Version

3.34.1

### What browsers are you seeing the problem on?

Firefox, Chrome, Safari

### What operating system are you using?

macOS

**Comment by snowystinger** [comment]
I just tried https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Elements/input/date in Safari 18.6 on my MacOS 15.6, but the VO+ArrowUp/Down did nothing except ding at me. Only the Arrow keys on their own worked.

I checked in our useSpinButton hook, key down isn't even called for the arrow when using VO+.
Is there a special setting I should have turned on?

From what I can see https://www.w3.org/WAI/ARIA/apg/patterns/spinbutton/ we are using the correct keys. I know these are VO specific ones, so trying to determine what is the right thing to do here. Are you able to use it with the arrow keys and not using the VO command?

**Comment by RavenAlly** [comment]
The W3C spinbutton example (https://www.w3.org/WAI/ARIA/apg/patterns/spinbutton/examples/datepicker-spinbuttons/) works as expected with the VO commands as described in the OP.

The big difference I can see between DateField and W3C implementation is that their spinbutton element has the keydown eventhandler applied directly while this DateField does not - perhaps VO cannot recognise DateField's keyboard handlers?

Caveat: Accessibility specialist, not an engineer, so take this with a grain of salt.

**Comment by snowystinger** [comment]
Thanks, I was able to reproduce. It appears to be an issue with contenteditable. If I apply that to the apg pattern example, it stops working and the keyboard event for the arrows never make it to the keydown handler. Not in bubbling nor capture phase. I'll check with out accessibility team and see if it's something they are aware of.

**Comment by majornista** [comment]
> The W3C spinbutton example (https://www.w3.org/WAI/ARIA/apg/patterns/spinbutton/examples/datepicker-spinbuttons/) works as expected with the VO commands as described in the OP.

In my tests of the [Datepicker SpinButton example in the Authoring Practices Guide](https://www.w3.org/WAI/ARIA/apg/patterns/spinbutton/examples/datepicker-spinbuttons/):

-  With Chrome version 140.0.7339.133, `Control` + `Option` + `ArrowDown`/`Up`, does not change the value, but as in the [React-Aria implementation](https://react-spectrum.adobe.com/react-aria/DateField.html), the `ArrowUp`/`ArrowDown` keys work without the  `Control` + `Option` modifiers. The same is true if you first drill into the stepper using `Control` + `Option` + `Shift` + `ArrowDown`, and then try to use `Control` + `Option` + `ArrowDown`/`Up`. 

-  With Safari version 18.6 (20621.3.11.11.3), `Control` + `Option` + `ArrowDown`/`Up`, does not change the value, but as in the [React-Aria implementation](https://react-spectrum.adobe.com/react-aria/DateField.html), the `ArrowUp`/`ArrowDown` keys work without the  `Control` + `Option` modifiers.  If you first drill into the stepper using `Control` + `Option` + `Shift` + `ArrowDown`, and then try to use `Control` + `Option` + `ArrowDown`/`Up`, it makes a difference and the value changes. 

- With Safari version 18.6 (20621.3.11.11.3), and the [React-Aria implementation](https://react-spectrum.adobe.com/react-aria/DateField.html), while the `ArrowUp`/`ArrowDown` keys work without the  `Control` + `Option` modifiers, they only beep when you try to use them with the `Control` + `Option` modifiers. If you first drill into the stepper using `Control` + `Option` + `Shift` + `ArrowDown`, and then try to use `Control` + `Option` + `ArrowDown`/`Up`, there is a bit of weirdness (at least on the first segment), because the `Control` + `Option` + `ArrowDown`/`Up` keys seem to increment or decrement the value of the focused segment and then immediately move focus to the next segment. The other segments do nothing on `Control` + `Option` + `ArrowDown`/`Up`.

I'm not sure that the native HTML5 date picker serves as a positive example. In my testing using the [native HTML5 date picker](https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Elements/input/date) with VoiceOver on both Chrome and Safari, the experience is much better if you disregard the VoiceOver instructions, and just use `ArrowDown`/`Up` without the `Control` + `Option` modifiers. In Chrome, value changes are announced inconsistently and are oftentimes interrupted by "beginning of text" or "end of text," leaving the user having to navigate away from and back to the segment to confirm the value change. In Safari, the behavior using `ArrowDown`/`Up` without the `Control` + `Option` modifiers is better, but adding the `Control` + `Option` modifiers does not allow you to change the value.
